proto/streamlit/proto/AutoRerun.proto

```diff
@@ -0,0 +1,25 @@
+/**!
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ syntax = "proto3";
+
+ message AutoRerun {
+   // The interval of reruns in seconds
+   float interval = 1;
+
+   // Optional: the fragment ID to rerun
+   string fragment_id = 2;
```

Comment:
If this is not provided, the full app is rerun, or? So we could reuse this to eventually provide the [streamlit-autorefresh](https://github.com/kmcgrady/streamlit-autorefresh) functionality from Ken natively in Streamlit?

Reply:
I think so -- at least I thought the same thing when porting this code over from the prototype. Note the comment already exists in the prototype, so I'm assuming you suspected the same thing while working on it a few months back 😆